parser/properties/properties_test.go

```diff
@@ -0,0 +1,38 @@
+package properties
+
+import (
+	"testing"
+)
+
+func TestPropertiesParser(t *testing.T) {
+	parser := &Parser{}
+	sample := `# This is a simle properties file
+    SAMPLE_KEY=https://example.com/
+! some comment=not-a-prop
+my-property=some-value`
+
+	var input interface{}
+	if err := parser.Unmarshal([]byte(sample), &input); err != nil {
+		t.Fatalf("parser should not have thrown an error: %v", err)
+	}
+
+	if input == nil {
+		t.Fatalf("there should be information parsed but its nil")
+	}
+
+	inputMap := input.(map[string]interface{})
+	myProp := inputMap["my-property"].(string)
+	if myProp != "some-value" {
+		t.Fatalf("Failed to parse property: %s", myProp)
```

Comment:
After the parsing, it makes sense to use `Errorf` instead of `Fatalf` so that if more than one of these checks fail, all issues will be shown rather than having only the first one to cause an issue to show.